overlord/configstate/config/helpers.go

```diff
@@ -279,8 +279,21 @@ type Conf interface {
 	State() *state.State
 }
 
+// ConfReader is an interface for reading of config values.
+type ConfReader interface {
+	Get(snapName, key string, result interface{}) error
+}
+
+// ApplyOptions encapsulates extra options passed to individual core config
+// handlers when configuration is applied to a specific root directory with
+// Apply().
+type ApplyOptions struct {
+	Preseeding bool
+	RootDir    string
+}
+
 // GetFeatureFlag returns the value of a given feature flag.
-func GetFeatureFlag(tr Conf, feature features.SnapdFeature) (bool, error) {
+func GetFeatureFlag(tr ConfReader, feature features.SnapdFeature) (bool, error) {
```

Comment:
I think this should ideally move to features.GetFlag but not an immediate concern,  but having ConfGetter defined here is strange, we can have a couple of copies of it anyway.

Reply:
we might want/need to introduce Transaction.GetMabye

Reply:
FWTW `GetFeatureFlag` takes `ConfGetter` as it applies "fuzzy" logic for true/false bools and "true"/"false" strings.

I'm a bit confused by the second comment; we already have `Transaction.GetMaybe` - is this a suggestion about simplifying this line from `GetFeatureFlag`:
```
if err := tr.Get(snapName, confName, &isEnabled); err != nil && !IsNoOption(err) {
		return false, err
...
```
with GetMaybe?

Reply:
yes, so ConfGetter would grow GetMaybe.  Sorry my point was to move this helper to features and  with GetMaybe we avoid importing overlord there if we do the move.

Reply:
to be clear it's follow-up material, but shouldn't be too hard. I maybe we can do a bit of prep here

Reply:
Added GetMaybe here, will do the rest in followups as discussed.